### PR TITLE
Do not render to/from itinerary addons with whitespace strings

### PR DESCRIPTION
**Pull request by grosbouddha**
Unit tested and tested locally in the storybook demo page by inserting temporary whitespace strings.

### CODE REVIEW

**Comment by Tuxosaurus** [comment]
Don't forget the changelog :)

Also, not sure if it is needed but the addon knobs in the story could be using text instead of booleans to reflect this fix.

File: src/itinerary/index.tsx [hunk 1]
```diff
@@ -27,12 +30,12 @@ const Itinerary = ({
 }: ItineraryProps) => {
   return (
     <ul className={cc([className, { 'kirk-itinerary--small': small }])}>
-      {headline && (
+      {isNonEmptyString(headline) && (
```
Comment (loick) [inline src/itinerary/index.tsx]: why not directly using !isEmpty? Since Typescript will enforce the usage of string anyway

Comment (grosbouddha) [reply src/itinerary/index.tsx]: To cover cases with strings made of whitespace spaces (e.g."   ").
I could rename isNonEmptyString to something more explicit to make it clear e.g. isEmptyOrWhitespace ?

Comment (loick) [reply src/itinerary/index.tsx]: Oh, right for the whitespace!

About the naming it's fine like that, thanks for the explanation :)

**Comment by grosbouddha** [comment]
> Don't forget the changelog :)
> 
> Also, not sure if it is needed but the addon knobs in the story could be using text instead of booleans to reflect this fix.

Added the storybook modification suggestions and the changelog entry, thanks!

**Comment by grosbouddha** [comment]
> Don't forget the changelog :)
> 
> Also, not sure if it is needed but the addon knobs in the story could be using text instead of booleans to reflect this fix.

Added the storybook modification suggestions and the changelog entry, thanks!

**Comment by grosbouddha** [comment]
PTAL

**Pull request merged**